src/yellowpaper/chapters/03-implementation/data-disclosure.tex

```diff
@@ -1,64 +1,83 @@
 \subsection{Data Disclosure}
 \label{sec:DataDisclosure}
-The process of data disclosure involves direct communication between a particular Educator, willing to disclose a part of the data, and an interested party~\tParty{B} (e. g. a recruiter), willing to pay for this data. Suppose an Educator~\tParty{E} has some data~$D$. To mitigate the risk of secondary market creation, one should ensure that the majority of the data remains in the private blocks. We propose to increase the cost of the data exponentially to its size, thus incentivizing the buyers to make as accurate requests as possible.
 
- Before the deal \tParty{E} ought to perform some preparation steps. \tParty{E} should:
+Disciplina architecture supports three types of data disclosure requests:
 \begin{enumerate}
-\item Divide~$D$ into $N$ chunks of size no more than 1~KiB:
+  \item Request for a set of authenticated private transactions satisfying some predicate (see details in Section \ref{sec:search-queries})
+  \item Request for personalization
+  \item Request for object disclosure
+\end{enumerate}
+
+Here we describe a protocol of fair data trade between the Educator as a seller and some interested party as a buyer. Despite a few variations the protocol is almost the same for all three types of the data disclosure requests. We first lay out the private transactions disclosure protocol. Then we describe modifications to the protocol so that one can apply it to other types of data.
+
+The process of data disclosure involves direct communication between a particular Educator~\tParty{E}, willing to disclose a part of the data, and an interested party~\tParty{B} (e. g. a recruiter), willing to pay for this data. Suppose \tParty{E} has some data~$D$. In case of private transactions $D$ is a set of \textit{authenticated transactions}, i.~e. tuples $(T_{priv}, P_{priv}, H, P_{pub})$. As shown in Section~\ref{sec:fair-cv} this data along with the educator's public key is enough to prove that a certain transaction $T_{priv}$ actually occurred in some private block of the given educator.
+
+The protocol fairness is guaranteed by a contract on the public chain. The contract is able to hold money and is stateful: it is capable of storing a log $L$ with data. All the data that parties send to the contract are appended to $L$.
+\begin{enumerate}
+\item The buyer \tParty{B} sends a signed search query $\Sign_\Party{B}(Q)$ directly to the seller \tParty{E}.
+\item Let $D$ be a set of authenticated transactions relevant for the query $Q$. \tParty{E} divides~$D$ into $N$ chunks of size no more than 1~KiB and computes a Merkle root of the plain data chunks:
```

Comment:
Wait, what for? I believe that this should be actually about fetching _existing_ proofs for data chunks, e. g. in case of basic search queries data chunks are transactions, and every transaction should be included into the existing block. So the seller should fetch all Merkle paths for transactions from _D_ (in form of _sparse Merkle forest_, I think), and send these paths along with encrypted data.
There is no point in computing a brand new Merkle tree over _unencrypted_ data. I think, this is misleading and should be changed.

Reply:
My bad, will fix

Reply:
> ... data ... are ...

Data is.

Reply:
I can argue with you on this one: https://language-boutique.com/lost-in-translation-full-reader/these-data-are-or-this-data-is-.html